src/raster/mdarray.rs

```diff
@@ -331,16 +331,10 @@ impl<'a> MDArray<'a> {
 
             let strings = string_pointers
                 .into_iter()
-                .map(|string_ptr| {
-                    if string_ptr.is_null() {
-                        String::new()
-                    } else {
-                        let string = _string(string_ptr);
-
-                        VSIFree(string_ptr as *mut c_void);
-
-                        string
-                    }
+                .filter_map(|ptr| {
+                    let s = _string(ptr);
+                    VSIFree(ptr as *mut c_void);
```

Comment:
ptr might here be actually null, free functions usually do not like to get a `nullptr` passed into them. Maybe better write

```
let s = _string(ptr)?;
VSIFree(ptr as *mut c_void);
Some(s)
```

Reply:
@rouault would you take a GDAL doc PR to explicitly mention that `VSIFree` can be called with a null pointer?